pkg/tsdb/sqleng/sql_engine.go

```diff
@@ -93,32 +124,32 @@ func (e *dataPlugin) transformQueryError(err error) error {
 	return e.queryResultTransformer.TransformQueryError(err)
 }
 
-// NewDataPlugin returns a new plugins.DataPlugin
-//nolint: staticcheck // plugins.DataPlugin deprecated
-func NewDataPlugin(config DataPluginConfiguration, queryResultTransformer SqlQueryResultTransformer,
-	macroEngine SQLMacroEngine, log log.Logger) (plugins.DataPlugin, error) {
-	plugin := dataPlugin{
+func NewQueryDataHandler(config DataPluginConfiguration, queryResultTransformer SqlQueryResultTransformer,
+	macroEngine SQLMacroEngine, log log.Logger) (*DataSourceHandler, error) {
```

Comment:
This could return the interface `backend.QueryDataHandler` instead of `*DataSourceHandler`

Reply:
why not :), we can return the interface, i am thinking it would be the same quantity of work behind the hook for go

Reply:
Was thinking because `DataSourceHandler` doesn't expose any exported fields, just the `QueryData` method which is what `backend.QueryDataHandler` defines. No biggie really.